README.md

```diff
@@ -1016,29 +1023,97 @@ cmake --build build
+* `OOMPH_WANT_MUMPS` (default `"OFF"`) requests that the parallel frontal solver MUMPS is built.
+* `OOMPH_WANT_HYPRE` (default `"OFF"`)  requests that the Hypre library is built.
+* `OOMPH_WANT_TRILINOS` (default `"OFF"`)  requests that the Trilinos library is built.
+* *** __Puneet:__ any others? I guess cgal (and friends), blas etc. are (or will be) built automatically since we can't really do without them; if so they only need to be discussed below when we exlain how to use existing versions of these libraries ***
 
-Since third party libraries are not modified/maintained by us, it is, of course, not necessary to rebuild them over and over again. They only have to be installed once in a permanent location, or they may already be installed somewhere on your computer. In this case, their location can be specified via variables such as 
+If any of these are set to `"ON"`, `CMake` will download the relevant sources and build the libraries as part of the `oomph-lib` installation. It will also (via the associated C++ macros, `OOMPH_HAS_MUMPS`, `OOMPH_HAS_HYPRE`, `OOMPH_HAS_TRILINOS`) enable the compilation of certain `oomph-lib` wrappers to these libraries. For instance, if `OOMPH_WANT_MUMPS` is set to `"ON"` during the project configuration, `oomph-lib`'s C++ sources will be compiled with the C++ macro `OOMPH_HAS_MUMPS` which then enables the creation of a `LinearSolver` object that uses `MUMPS` as the underlying linear solver.
+
+Since third party libraries are not modified/maintained by us, it is, of course, not necessary to rebuild them over and over again. They only have to be built/installed once (ideally using full optimisation) in a permanent location, or they may already be installed somewhere on your computer. In this case, their location can be specified via the following variables:
 * `OOMPH_USE_BLAS_FROM` which specifies the absolute path to the directory in which the BLAS library has been installed; this directory should contain an `include` and `lib` subdirectory *** __Puneet:__ Making this up as I go along; we ought to specify what exactly the path is to ***.
 * ** __Puneet:__ I assume there are others; list/describe here. **
 
 
 
-*** MH TO CONTINUE WHEN THERE'S A FULL LIST OF SET-ABLE CACHE VARIABLES ***
+Finally, there are a number of cache variables that are only of interest to `oomph-lib` developers and can/should be ignored. A full list of cache variables can be produced by `CMake` but sadly only once the project has been configured. Luckily the configuration is quick, so to get the full list simply configure the project without providing any variables and obtain a list of variables afterwards. The re-run the configuration if you want to use any newly discovered variables:
+```bash
+# Default configuration
+cmake -G ninja -B build
+
+# Have a look at the available configure options
+cd build
+cmake -LH
+```
+
+*** __Puneet:__ I think this is what you (and google!) said. However, I get a warning and an error when I do this:
```

Comment:
Although I prefer your approach as it allows you inspect variables in whatever configuration you have

Reply:
Ha! I tried that (because I'd noticed the two dots after the command but that doesn't work either:

```bash
matthias@matthias-Precision-7730:~/oomph-lib$ cd build/
matthias@matthias-Precision-7730:~/oomph-lib/build$ cmake -LH ..
# ------------------------------------------------------------------------------
#                                          888             888 d8b 888
#                                          888             888 Y8P 888
#                                          888             888     888
#  .d88b.   .d88b.  88888b.d88b.  88888b.  88888b.         888 888 88888b.
# d88""88b d88""88b 888 "888 "88b 888 "88b 888 "88b        888 888 888 "88b
# 888  888 888  888 888  888  888 888  888 888  888 888888 888 888 888  888
# Y88..88P Y88..88P 888  888  888 888 d88P 888  888        888 888 888 d88P
#  "Y88P"   "Y88P"  888  888  888 88888P"  888  888        888 888 88888P"
#                                 888
#                                 888
#                                 888
# ------------------------------------------------------------------------------
CMake Error at cmake/OomphConfigureProject.cmake:11 (message):
  In-source builds not allowed.  Please make a new directory (called a build
  directory) and run CMake from there.
Call Stack (most recent call first):
  CMakeLists.txt:60 (include)


-- Configuring incomplete, errors occurred!
See also "/home/matthias/oomph-lib/CMakeFiles/CMakeOutput.log".
-- Cache values
// 
CMAKE_BUILD_TYPE:STRING=Release

// Install path prefix, prepended onto install directories.
CMAKE_INSTALL_PREFIX:PATH=/usr/local
```

Reply:
That's very odd and certainly doesn't happen when I do it. We should discuss this when we next have a call...